contracts/MarginBase.sol

```diff
@@ -272,6 +276,24 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
             // define market via _marketKey
             IFuturesMarket market = futuresMarket(_newPositions[i].marketKey);
 
+            /// @notice (if it exists) update position here to ensure internal
+            /// accounting is up-to-date moving forward
+            /// @dev if position has been liquidated, this contract will
+            /// effectively create a new position as specified by _newPositions[i]
+            if (activeMarketPositions[_newPositions[i].marketKey].size != 0) {
```

Comment:
let's cache `activeMarketPositions[_newPositions[i].marketKey].size` as `activePositionSize` locally not only for gas, but mainly readability

Reply:
It seems like that value is being used twice, but they might be different

ex: if a given position was liquidated, then on line 283 `activeMarketPositions[_newPositions[i].marketKey].size` will be non-zero (bc it is out-of-date). After calling `fetchPositionAndUpdate()` the size of that same position will then become zero, and therefore, the second check on line 291 `activeMarketPositions[_newPositions[i].marketKey].size` will actually be different (i.e. zero). 

Furthermore, labeling that value as `activePositionSize` may actually be misleading IF the position was liquidated due to the fact that the position is not technically active (the mapping returns the default value for `size` after the key has been deleted). 

One consideration is if we want the second check at all... Worth considering...